app/helpers/sheet/group/nav_left.rb

```diff
@@ -31,8 +31,31 @@ def render
 
       private
 
+      def sort_list_by_name(objects)
+        children_of = {}
+        objects.each do |o|
+          children_of[o.parent_id] ||= []
+          children_of[o.parent_id] << o
+        end
+
+        children_of.each_value do |children|
+          children.sort_by! { |c| c.short_name || c.name }
+        end
+
+        results = []
+        recombine_lists(results, children_of, objects.first.parent_id)
+        results
+      end
+
+      def recombine_lists(results, children_of, parent_id)
+        children_of[parent_id]&.each do |o|
+          results << o
+          recombine_lists(results, children_of, o.id)
+        end
+      end
```

Comment:
Mostly inspired by https://github.com/collectiveidea/awesome_nested_set/wiki/How-to-generate-nested-unordered-list-tags-with-one-DB-hit

Should I put these in a separate helper?

Reply:
Resolved offline: Not now. If something similar is needed we can still extract it.